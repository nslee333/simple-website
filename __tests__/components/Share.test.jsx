import { render, screen } from "@testing-library/react";
import { toHaveAttribute } from "@testing-library/jest-dom";
import Share from "../../src/components/Share";
import renderer from "react-test-renderer";

test("Share renders without crashing", () => {
  render(<Share />);
});

test("Share renders correctly according to snapshot", () => {
  const tree = renderer.create(<Share />).toJSON();

  expect(tree).toMatchSnapshot();
});

test("Contact Us button has a href value of #contact", () => {
  render(<Share />);

  expect(screen.getByRole("link")).toHaveAttribute("href", "#contact");
});
