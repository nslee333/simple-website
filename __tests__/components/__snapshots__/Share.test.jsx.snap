// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Share renders correctly according to snapshot 1`] = `
<div
  className="m-5  h-auto  w-[42rem]  min-w-[18.85rem]  rounded-md  bg-white  p-5  sm:h-[25rem]"
>
  <h3
    className="text-3xl italic xs:mt-2"
  >
    Have something to share?
  </h3>
  <h4
    className="text-2xl italic xs:mt-10"
  >
    We're always looking for speakers
  </h4>
  <div
    className="italic xs:mt-10"
  >
    If you have something to share, big or small, big or small, please get in touch!
  </div>
  <div
    className="mb-2 mt-10 sm:mt-32 hover:translate-y-0.5"
  >
    <a
      className="shadow-grey  rounded-lg  bg-[#7FE7D7]  p-3  text-2xl  shadow  hover:origin-bottom  hover:shadow-none"
      href="#contact"
      htmlFor="button"
    >
      Contact Us
    </a>
  </div>
</div>
`;
